contracts/RewardManager.sol

```diff
@@ -224,6 +254,14 @@ contract RewardManager is Ownable, Versionable, Safe {
       abi.encodeWithSelector(SWAP_OWNER, address(this), oldOwner, newOwner);
   }
 
+  function getWithdrawFromRewardSafeData(
```

Comment:
@alex-cardstack I think you can just pass the data in from offchain. Make a `data` param for this function. Mainly because I think the data for a transfer function is clear for ERC677 or ERC20 is unambiguous. Saves contract space. `getTransferRewardSafeData` saves client from knowing what the contract is. Using that logic, can you perhaps replace [this](https://github.com/cardstack/card-pay-protocol/blob/f145e7a4d1277e388ff71cfb1e1199e23c5b9836/test/utils/helper.js#L1298) with a smart contract call.

Reply:
Think this would save us contract space.

Reply:
Would like to hear your opinon on this? Becos in truth, we can also remove `getTransferRewardSafeData` honestly

Reply:
@habdelra tagging you here because this has been a pattern for `transfer` in prepaid card manager also. I was wondering if there was an actual reason why `getTransferCardData` is written on chain when in truth the client can figure that out offchain and generate the data offchain.

Reply:
I think this is a relic from the original prepaidcard manager that Tim’s team worked on. If we aren’t calling this onchain anymore, then yeah we can remove it.

Reply:
@alex-cardstack then can you get rid of `getTrasferRewardSafeData` and just make everything offchain. I just checked in the linked list implementation, so it is as expected that `prevOwner` can remain as the contract address and the `oldOwner` and `newOwner` are the only thing changing.

Reply:
@tintinthong I think we are both too laszer focused on the prepaidCardManager logic--`getWithdrawFromRewardSafeData` is actually used on chain--it's in the execTransaction herehttps://github.com/cardstack/card-pay-protocol/pull/131/files#diff-d99035ba0a6829c3a3a5f7ee88c5df2590f4c95c550100a03b81186e0e1f512aR226. But yes, we do have some left over stuff that in the PrepaidCardManager is no longer in the `execTransaction` after the `send()` that we did on it awhile back.

Reply:
what do you mean? my proposal is to just remove the onchain function that calls for data such as `getWithdrawFromRewardSafeData` and `getTransferRewardSafeData` and pass the data in as an argument to the main function, such as `withdrawFromSafe` and `transferRewardSafe`. We will not touch the prepaid card manager bcos that is critical path.

Reply:
@tintinthong had a call with @habdelra this morning, and he thinks you and he were thinking of the prepaid card manager send function, however in this case it does need to be on-chain because otherwise the client can send any transaction to the token that the contract will blindly sign - it being on-chain here is what verifies the contract owner and EOA owner actually agree on the same action

Reply:
@tintinthong oh my misunderstanding--I was the one with PrepaidCardManager on the brain then. I see what you mean. 

@alex-cardstack in this case the `withdrawFromRewardSafe` itself would have this data (the data that is returned from `getWithdrawFromRewardSafeData` as derivable with web3 or ethers) passed in as an additional param from what you already have,

Reply:
Actually @habdelra I take everything back. @alex-cardstack I think is right.

The client can piggy back the `withdrawTransferFunction` to send whatever data he wants other than TRANSFER. So this thing has to be onchain.

Reply:
Sorry for the confusion then. It seems that `getWithdrawFromRewardSafeData` and `getTransferRewardSafeData` all have to be functions on-chain.

Reply:
However, neither of these need to be public view functions at all - they are only used in a single place, I think it would save gas to just inline both `getWithdrawFromRewardSafeData` and `getTransferRewardSafeData`, they can both be a single line in the calling functions